strapi/strapi
# Fix suppressed policy errors

### What does it do?

Allows UnauthorizedError and ForbiddenError in the auth middleware to bubble up rather than catching them and sending our own empty error

TODO:

- [ ] write test that Policy throwing a PolicyError should bubble up

### Why is it needed?

PolicyError thrown in policies is currently being suppressed and returning a generic ForbiddenError instead

### How to test it?

- create a policy
- have it throw a PolicyError

- without this PR: see that the api currently returns a ForbiddenError instead
- using this PR: it returns the correct PolicyError

Tests are in progress

### Related issue(s)/PR(s)

Fixes #18566
